### PR TITLE
Use serde_json APIs instead relying on value enum

### PR DESCRIPTION
**Pull request by macisamuele**
The goal of this PR is to decouple (as much as possible) from `serde_json`.
This is mostly needed to simplify the integration of `json-trait-rs` (#30).

I'm publishing this for initial review as it will be the base of the work for testing `json-trait-rs`.

I'm very sorry for the size of the PR ... :(

### CODE REVIEW

File: src/keywords/helpers.rs [hunk 1]
```diff
@@ -1,8 +1,19 @@
 use serde_json::Value;
 
+fn are_f64_equal(left: f64, right: f64) -> bool {
+    (left - right).abs() < 3. * f64::EPSILON
```
Comment (Stranger6667) [inline src/keywords/helpers.rs:4]: Why not comparison of `left_num` and `right_num`?

**Comment by Stranger6667** [comment]
> I'm very sorry for the size of the PR ... :(

No probs :) 

I see that it does some negative impact on performance (however it is small in absolute numbers), e.g. with some `type` benches it has >+100% time change. I understand that it might be a cost to support non-serde case, but is there any way to avoid this? Or what ways do we have to improve it in the future?

**Comment by macisamuele** [comment]
I was noticing the performance hit, but overall absolute number were very low.

Still I'm not 100% sure that json-trait-ts integration is feasible and reasonable. So I would eventually suggest to not merge this yet until some further tests are done

**Comment by Stranger6667** [comment]
Yep, let's take a look at how it could work performance-wise in comparison with other approaches.
I opened drafts #53 and #54 for comparison (I'd be glad if you can take a look). Will try to clean them up so it could be measured better

Cheers

**Comment by macisamuele** [comment]
@Stranger6667 I'm going to drop this for now as it does have a massive amount of merge conflicts as well as #86 might also simplify this effort as it would require a very localised diff and not on all the keywords.

**Pull request closed without merging**